src/Compilers/CSharp/Test/Semantic/Semantics/PatternMatchingTests.cs

```diff
@@ -7364,6 +7364,1563 @@ static void Test()
+                    // (9,13): error CS0656: Missing compiler required member 'System.MemoryExtensions.AsSpan'
+                    //             "string 1" => 1,
+                    Diagnostic(ErrorCode.ERR_MissingPredefinedMember, @"""string 1""").WithArguments("System.MemoryExtensions", "AsSpan").WithLocation(9, 13),
+                    // (10,13): error CS0656: Missing compiler required member 'System.MemoryExtensions.SequenceEqual'
+                    //             "string 2" => 2,
+                    Diagnostic(ErrorCode.ERR_MissingPredefinedMember, @"""string 2""").WithArguments("System.MemoryExtensions", "SequenceEqual").WithLocation(10, 13),
+                    // (10,13): error CS0656: Missing compiler required member 'System.MemoryExtensions.AsSpan'
+                    //             "string 2" => 2,
+                    Diagnostic(ErrorCode.ERR_MissingPredefinedMember, @"""string 2""").WithArguments("System.MemoryExtensions", "AsSpan").WithLocation(10, 13));
+        }
+
+        [Fact]
+        public void PatternMatchSpanCharOnConstantStringCSharp_8()
+        {
+            var source =
+@"
+using System;
+class C
+{
+    static bool M(Span<char> chars) => chars is """";
+}
+";
+            CreateCompilationWithSpanAndMemoryExtensions(source, parseOptions: TestOptions.Regular8)
+                .VerifyDiagnostics(
+                    // (5,49): error CS8652: The feature 'pattern matching ReadOnly/Span<char> on constant string' is currently in Preview and *unsupported*. To use Preview features, use the 'preview' language version.
+                    //     static bool M(Span<char> chars) => chars is "";
+                    Diagnostic(ErrorCode.ERR_FeatureInPreview, @"""""").WithArguments("pattern matching ReadOnly/Span<char> on constant string").WithLocation(5, 49));
+        }
+
+        [Fact]
+        public void PatternMatchSpanCharOnNull()
+        {
+            var source =
+@"
+using System;
+class C
+{
+    static bool M(Span<char> chars) => chars is null;
+}";
+            CreateCompilationWithSpanAndMemoryExtensions(source, parseOptions: TestOptions.RegularPreview)
+                .VerifyDiagnostics(
+                    // (5,49): error CS0150: A constant value is expected
+                    //     static bool M(Span<char> chars) => chars is null;
+                    Diagnostic(ErrorCode.ERR_ConstantExpected, "null").WithLocation(5, 49));
+        }
+
+        /// <summary>
+        /// This has a different diagnostic to <see cref="SwitchReadOnlySpanCharOnNull"/> because there is a valid implicit conversion from null to Span via Array,
+        /// but not to the ReadOnlySpan defined in <see cref="TestSources.Span"/> which has two implicit operators from reference types to ReadOnlySpan (for string and array).
+        /// When using the BCL, the diagnostic will be this one for both.
+        /// </summary>
```

Comment:
We should test with the BCL types rather than TestSources, to avoid surprises (due to different implicit conversions, etc.) when compiling projects that are targeting the BCL.

Please consider changing the new tests in this file that depend on definitions of  `System.Span<T>` and `System.MemoryExtensions` to compile against the BCL instead. Since those types require CoreCLR, the affected tests will need to use `[ConditionalFact]`.

```csharp
[ConditionalFact(typeof(CoreClrOnly))]
public void SwitchSpanCharOnNull()
{
    var source = "...";
    CreateCompilation(sourceA, targetFramework: TargetFramework.NetCoreApp, parseOptions: TestOptions.RegularPreview);
    // ...
}
```